android/TritonClient/tritonlib/src/main/java/com/mobiledgex/tritonlib/ImageSenderTriton.java

```diff
@@ -0,0 +1,280 @@
+            this.activity = activity;
+            return this;
+        }
+
+        public Builder setImageServerInterface(ImageServerInterface imageServerInterface) {
+            this.imageServerInterface = imageServerInterface;
+            return this;
+        }
+
+        public Builder setCloudLetType(ImageServerInterface.CloudletType cloudLetType) {
+            this.cloudLetType = cloudLetType;
+            return this;
+        }
+
+        public Builder setTls(boolean tls) {
+            this.tls = tls;
+            return this;
+        }
+
+        public Builder setHost(String host) {
+            this.host = host;
+            return this;
+        }
+
+        public Builder setModelName(String modelName) {
+            this.modelName = modelName;
+            return this;
+        }
+
+        public Builder setPort(int port) {
+            this.port = port;
+            return this;
+        }
+
+        public Builder setCameraMode(CameraMode cameraMode) {
+            this.cameraMode = cameraMode;
+            return this;
+        }
+
+        public ImageSenderTriton build() {
+            return new ImageSenderTriton(this);
+        }
+    }
+
+    private ImageSenderTriton(final Builder builder) {
+        Log.i(TAG, "mLatencyFullProcessRollingAvg="+mLatencyFullProcessRollingAvg);
+        mCloudLetType = builder.cloudLetType;
+        mTls = builder.tls;
+        mHost = builder.host;
+        mPort = builder.port;
+        mModelName = builder.modelName;
+        mContext = builder.activity;
+        setCameraMode(builder.cameraMode);
+
+        mImageServerInterface = builder.imageServerInterface;
+
+        Log.i(TAG, "ImageSenderTriton "+mCloudLetType+" "+mHost+":"+mPort+" mTls="+mTls+" mCameraMode="+mCameraMode);
+        super.init();
+        Log.i(TAG, "mConnectionMode="+mConnectionMode);
+    }
+
+    /**
+     * Encode the bitmap and use Volley async to request face detection/recognition
+     * coordinates. Decode the returned JSON string and update the rectangles
+     * on the preview. Also time the transaction to calculate latency.
+     *
+     * @param bitmap  The image to encode and send.
+     */
+    public void sendImage(Bitmap bitmap) {
+        Log.d(TAG, mCloudLetType+" sendImage()");
+        if(mBusy || mInactive || mInactiveBenchmark || mInactiveFailure) {
+            return;
+        }
+
+        // Get a lock for the busy
+        mBusy = true;
+        if(mDoNetLatency) {
+            if (mHandler == null) {
+                // Too soon.
+                return;
+            }
+            mHandler.post(new Runnable() {
+                @Override
+                public void run() {
+                    doSinglePing(mHost, mLatencyNetOnlyRollingAvg, mCloudLetType);
+                }
+            });
+        }
+
+        ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+        bitmap.compress(Bitmap.CompressFormat.JPEG, 67, byteStream);
```

Comment:
Is 67 coming from a Triton guideline or this is a best practice by experience?

Reply:
Sweet spot for size/accuracy on existing CV server. Could  stand to do some testing with Triton.